src/overlays/ovl_i6/fox_ve2.c

```diff
@@ -0,0 +1,4132 @@
+            Math_SmoothStepToF(&D_80177A48[1], 100.0f, 1.0f, 10.0f, 0.0f);
+            Math_SmoothStepToF(&gPlayer[0].camEye.x, boss->obj.pos.x + sp64.x, 0.02f, D_80177A48[0], 0.0f);
+            Math_SmoothStepToF(&gPlayer[0].camEye.y, boss->obj.pos.y + sp64.y, 0.02f, D_80177A48[0], 0.0f);
+            Math_SmoothStepToF(&gPlayer[0].camEye.z, boss->obj.pos.z + sp64.z, 0.02f, D_80177A48[0], 0.0f);
+            Math_SmoothStepToF(&gPlayer[0].camAt.x, boss->obj.pos.x, 0.05f, D_80177A48[1], 0.0f);
+            Math_SmoothStepToF(&gPlayer[0].camAt.y, boss->obj.pos.y, 0.05f, D_80177A48[1], 0.0f);
+            Math_SmoothStepToF(&gPlayer[0].camAt.z, boss->obj.pos.z, 0.05f, D_80177A48[1], 0.0f);
+            Math_SmoothStepToF(&gPlayer[0].unk_034, 0.0f, 0.5f, 0.5f, 0.0f);
+
+            if (D_80177A80 == 50) {
+                func_800BA808(gMsg_ID_19335, RCID_ANDROSS_RED);
+            }
+            if (D_80177A80 == 200) {
+                D_80178284 = 0;
+                gPlayer[0].unk_0D0 = D_80161A54;
+                func_8001CA24(0);
+                boss->actionState = 21;
+                func_8007B344(boss->obj.pos.x, boss->obj.pos.y, boss->obj.pos.z, 60.0f, 5);
+                func_8007A568(boss->obj.pos.x, boss->obj.pos.y, boss->obj.pos.z, 40.0f);
+
+                for (i = 0; i < ARRAY_COUNT(gActors); i++) {
+                    if ((gActors[i].obj.status == 2) && (gActors[i].obj.id == OBJ_ACTOR_286)) {
+                        Object_Kill(&gActors[i].obj, &gActors[i].sfxPos);
+                    }
+                }
+                D_80177AB0 = 4;
+                D_80177A80 = 0;
+            }
+            break;
+        case 21:
+            func_i6_801876FC();
+            D_80177A80++;
+            D_80178480 = 10;
+            if (D_80177A80 < 200) {
+                D_80178380[0] += 4;
+                if (D_80178380[0] >= 0xFF) {
+                    D_80178380[0] = 0xFF;
+                }
+                D_80178390[0] = D_801783A0[0] = D_801783B0[0] = 0xFF;
+                Math_SmoothStepToF(&D_800CA230, 0.15f, 0.2f, 0.004f, 0.0f);
+                Math_SmoothStepToF(&D_801779A8[gMainController], 70.0f, 1.0f, 4.0f, 0.0f);
+            }
+            switch (D_80177A80) {
+                case 30:
+                    gRadioState = 0;
+                    func_800BA808(gMsg_ID_20318, RCID_FOX);
+                    gPlayer[0].timer_220 = 200;
+                    break;
+                case 350:
+                    func_800BA808(gMsg_ID_19468, RCID_JAMES);
+                    break;
+                case 500:
+                    func_800BA808(gMsg_ID_19350, RCID_FOX);
+                    D_8015F924 = 1;
+                    break;
+                case 600:
+                    D_80177AB0 = 5;
+                    gFogRed = 0x81;
+                    gFogGreen = 0x1B;
+                    gFogBlue = 0;
+                    gFogNear = 0x3E4;
+                    gFogFar = 0x3EF;
```

Comment:
```suggestion
                    gFogRed = 129;
                    gFogGreen = 27;
                    gFogBlue = 0;
                    gFogNear = 996;
                    gFogFar = 1007;
```
Colors and fog parameters in decimal. RGBA16 colors like gBgColor can stay in hex.